worldwind/src/commonMain/kotlin/earth/worldwind/render/program/GeomLinesShaderProgram.kt

```diff
@@ -0,0 +1,216 @@
+package earth.worldwind.render.program
+
+import earth.worldwind.draw.DrawContext
+import earth.worldwind.geom.Matrix3
+import earth.worldwind.geom.Matrix4
+import earth.worldwind.geom.Vec2
+import earth.worldwind.geom.Viewport
+import earth.worldwind.render.Color
+import earth.worldwind.util.kgl.KglUniformLocation
+
+open class GeomLinesShaderProgram : AbstractShaderProgram() {
+    override var programSources = arrayOf(
+        """
+            uniform mat4 mvpMatrix;
+            uniform float lineWidth;
+            uniform vec2 screen;
+            uniform bool enableTexture;
+            uniform bool enableOneVertexMode;
+            uniform mat3 texCoordMatrix;
+            
+            attribute vec4 pointA;
+            attribute vec4 pointB;
+            attribute vec4 pointC;
+            attribute vec2 vertexTexCoord;
+            
+            varying vec2 texCoord;
+            
+            void main() {
+                if(enableOneVertexMode)
+                {
+                    /* Transform the vertex position by the modelview-projection matrix. */
+                    gl_Position = mvpMatrix * pointA;
```

Comment:
Why we assume here w==1, but in else use explicit value?

Reply:
Enable one vertex mode uses only 3 floats for pointA vertex attrib, seems like in this case w is defined as 1. On the other hand in else mode we use whole 4 floats of attribute and data there won't be equal to 1 hence explicit value. Probably safer to use explicit value here also to not rely on undocumented behaviour